pkg/registry/deployment/etcd/etcd.go

```diff
@@ -184,53 +187,76 @@ type ScaleREST struct {
-// 	}
-// 	scale, err := extensions.ScaleFromDeployment(deployment)
-// 	if err != nil {
-// 		return nil, errors.NewBadRequest(fmt.Sprintf("%v", err))
-// 	}
-// 	return scale, nil
-// }
-
-// func (r *ScaleREST) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool, error) {
-// 	if obj == nil {
-// 		return nil, false, errors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
-// 	}
-// 	scale, ok := obj.(*extensions.Scale)
-// 	if !ok {
-// 		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
-// 	}
-
-// 	if errs := extvalidation.ValidateScale(scale); len(errs) > 0 {
-// 		return nil, false, errors.NewInvalid(extensions.Kind("Scale"), scale.Name, errs)
-// 	}
-
-// 	deployment, err := (*r.registry).GetDeployment(ctx, scale.Name)
-// 	if err != nil {
-// 		return nil, false, errors.NewNotFound(extensions.Resource("deployments/scale"), scale.Name)
-// 	}
-// 	deployment.Spec.Replicas = scale.Spec.Replicas
-// 	deployment, err = (*r.registry).UpdateDeployment(ctx, deployment)
-// 	if err != nil {
-// 		return nil, false, errors.NewConflict(extensions.Resource("deployments/scale"), scale.Name, err)
-// 	}
-// 	newScale, err := extensions.ScaleFromDeployment(deployment)
-// 	if err != nil {
-// 		return nil, false, errors.NewBadRequest(fmt.Sprintf("%v", err))
-// 	}
-// 	return newScale, false, nil
-// }
+// ScaleREST implements Patcher
+var _ = rest.Patcher(&ScaleREST{})
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &autoscaling.Scale{}
+}
+
+func (r *ScaleREST) Get(ctx api.Context, name string) (runtime.Object, error) {
+	deployment, err := (*r.registry).GetDeployment(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound(autoscaling.Resource("deployments/scale"), name)
+	}
+	scale, err := scaleFromDeployment(deployment)
+	if err != nil {
+		return nil, errors.NewBadRequest(fmt.Sprintf("%v", err))
+	}
+	return scale, nil
+}
+
+func (r *ScaleREST) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	if obj == nil {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
+	}
+	scale, ok := obj.(*autoscaling.Scale)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
+	}
+
+	if errs := asvalidation.ValidateScale(scale); len(errs) > 0 {
+		return nil, false, errors.NewInvalid(autoscaling.Kind("Scale"), scale.Name, errs)
+	}
+
+	deployment, err := (*r.registry).GetDeployment(ctx, scale.Name)
+	if err != nil {
+		return nil, false, errors.NewNotFound(autoscaling.Resource("deployments/scale"), scale.Name)
+	}
+	deployment.Spec.Replicas = scale.Spec.Replicas
+	deployment, err = (*r.registry).UpdateDeployment(ctx, deployment)
+	if err != nil {
+		return nil, false, err
+	}
+	newScale, err := scaleFromDeployment(deployment)
+	if err != nil {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("%v", err))
```

Comment:
Is throwing an error the correct thing to do in this case?  We have just successfully updated the deployment with the new scale, but can't convert the deployment back to a scale because the label selectors are invalid on the deployment.  Is the deployment corrupted if the label selector is invalid?